$NetBSD: patch-ac,v 1.27 2013/01/03 15:37:23 ryoon Exp $

First chunk:
Add support for thread limits.
From Peter Stephenson <Peter.Stephenson@csr.com> on zsh-workers.

All other chunks:
Make this package build on Interix.  Patches provided by Hiramatsu
Yoshifumi in PR pkg/25946.
Last chunk only:
Don't force -Wl,--whole-archive for NetBSD, it breaks the build with
clang, which doesn't have a reason to disable it again.

All chunks:
Accepted by Peter Stephenson <p.w.stephenson@ntlworld.com> on
zsh-workers on 2012-08-16.

--- configure.orig	2012-09-15 20:13:14.000000000 +0000
+++ configure
@@ -9645,6 +9645,42 @@ if test $zsh_cv_have_RLIMIT_NTHR = yes; 
 
 fi
 
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for limit RLIMIT_NTHR" >&5
+$as_echo_n "checking for limit RLIMIT_NTHR... " >&6; }
+if ${zsh_cv_have_RLIMIT_NTHR+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+#include <sys/types.h>
+#ifdef HAVE_SYS_TIME_H
+#include <sys/time.h>
+#endif
+#include <sys/resource.h>
+int
+main ()
+{
+RLIMIT_NTHR
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_compile "$LINENO"; then :
+  zsh_cv_have_RLIMIT_NTHR=yes
+else
+  zsh_cv_have_RLIMIT_NTHR=no
+fi
+rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $zsh_cv_have_RLIMIT_NTHR" >&5
+$as_echo "$zsh_cv_have_RLIMIT_NTHR" >&6; }
+
+if test $zsh_cv_have_RLIMIT_NTHR = yes; then
+  $as_echo "#define HAVE_RLIMIT_NTHR 1" >>confdefs.h
+
+fi
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for limit RLIMIT_NOFILE" >&5
 $as_echo_n "checking for limit RLIMIT_NOFILE... " >&6; }
 if ${zsh_cv_have_RLIMIT_NOFILE+:} false; then :
@@ -11630,6 +11666,10 @@ $as_echo "$zsh_cv_sys_elf" >&6; }
         DLLD="${DLLD=$CC}"
         DLLDARG=""
       ;;
+      *interix*)
+        DLLD="${DLLD=$CC}"
+        DLLDARG=""
+      ;;
       * )
 	DLLD="${DLLD=ld}"
 	DLLDARG=""
